### DISCUSSION TITLE
Reconsider the need for ColumnarDataSource

### DISCUSSION THREAD

**Discussion by p-himik** [bokeh/bokeh#11562] [Internals and design (Q&A)]
In the process of thinking about #11560 I found that `ColumnarDataSource` is useful only for `GeoJSONDataSource` right now and has been introduced in #5642 as a band-aid to not delay the release.

It's been quite some time since then - perhaps the band-aid should be turned into a more proper solution?

**Comment by bryevdv** [comment]
> It's been quite some time since then - perhaps the band-aid should be turned into a more proper solution?

That would definitely been welcome. Had the CZI R4 funding come through, half the proposal was dedicated to GIS related improvements, and would have been a natural way to jump start work in this area. Unfortunately the proposal was not selected. So, if you have concrete ideas or sketches please record them somewhere :) 

The basic idea behind `GeoJSONDataSource` was to be able to adapt GeoJSON inputs into a CDS-like format that all glyphs could just use directly (not so different from filtering or transforming I suppose but there are performance considerations to avoid re-converting on every use, etc.)

**Comment by p-himik** [comment]
I see. It might tie in nicely to our other discussion about a filtering data source wrapper by allowing arbitrary transformations. The performance should be a non-issue if wrappers cache their results. We'll see how my experiments go.
